backend/src/main/java/trainingjournal/backend/controller/UserController.java

```diff
@@ -15,27 +15,48 @@ public class UserController {
 
     private final UserService userService;
 
-    public UserController(UserService userService){
+    public UserController(UserService userService) {
         this.userService = userService;
     }
 
     @GetMapping("/me")
-    public String helloMe(Principal principal){
-        if(principal != null){
+    public String helloMe(Principal principal) {
+        if (principal != null) {
             return principal.getName();
         }
         return "anonymous";
     }
 
     @PostMapping("/login")
-    public String login(){
+    public String login() {
         return SecurityContextHolder.getContext().getAuthentication().getName();
     }
 
     @PostMapping("{username}/exercises/")
-    public Set<Exercise> addUserExercisesList(@PathVariable String username, @RequestBody List<Exercise> exercisesList){
+    public Set<Exercise> addUserExercisesList(@PathVariable String username, @RequestBody List<ExerciseDTO> exercisesList) {
 
         return userService.addUserExercisesList(username, exercisesList);
     }
 
+    @PostMapping("/newuser/")
+    public GymUser addNewGymUser(@RequestBody GymUserSignup signupData){
+
+        return userService.addNewGymUser(signupData);
+    }
+
+    @PutMapping("{username}/personaldata/updateusername/")
+    public GymUser updateUsername(@PathVariable String username,
+                                          @RequestBody String newUsername
+
+    ) {
+        return userService.updateUsername(username,newUsername);
+    }
+
+    @PutMapping("{username}/personaldata/updateweight/")
```

Comment:
Hier das gleiche bei updateusername -> PUT `/api/users/{username}/` würde heißen, dass Daten von {username} verändert werden -> somit ist auch keine separate Methode für das Gewicht notwendig, weil das Gewicht ein Teil der Userdaten ist.
Wenn man unbedingt einen Endpunkt dafür haben möchte, könnte man
PUT `/api/users/{username}/weight` erstellen


(Hatten wir das in der Planung nicht besprochen? :smiley: )